src/main/java/bio/terra/workspace/service/datareference/flight/CreateDataReferenceFlight.java

```diff
@@ -2,20 +2,19 @@
 
 import bio.terra.stairway.Flight;
 import bio.terra.stairway.FlightMap;
-import bio.terra.workspace.db.DataReferenceDao;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import org.springframework.context.ApplicationContext;
+import bio.terra.workspace.common.utils.FlightApplicationContext;
 
 public class CreateDataReferenceFlight extends Flight {
 
   public CreateDataReferenceFlight(FlightMap inputParameters, Object applicationContext) {
     super(inputParameters, applicationContext);
 
-    ApplicationContext appContext = (ApplicationContext) applicationContext;
-    DataReferenceDao dataReferenceDao = (DataReferenceDao) appContext.getBean("dataReferenceDao");
-    ObjectMapper objectMapper = (ObjectMapper) appContext.getBean("objectMapper");
+    FlightApplicationContext appContext =
```

Comment:
So is this constructor constrained somehow such that you can't simply wire in `dataReferenceDao` and `objectMapper`? There's also field autowiring, but I've read that's frowned upon.

Reply:
Spring only autowires classes that are @Component or similar - they are singletons.
Things that  you `new` cannot use autowire.

Reply:
oh right, flights aren't singletons.